core/src/avm2/globals/flash/system/System.as

```diff
@@ -15,5 +16,24 @@ package flash.system {
         public static function disposeXML(node:XML):void {
             stub_method("flash.system.System", "disposeXML");
         }
+
+        public static function get freeMemory(): Number {
+            stub_getter("flash.system.System", "freeMemory");
+            return 1024*1024*10;
+        }
+
+        public static function get privateMemory(): Number {
+            stub_getter("flash.system.System", "privateMemory");
+            return 1024*1024*100;
+        }
+
+        public static function get totalMemoryNumber(): Number {
+            stub_getter("flash.system.System", "totalMemoryNumber");
+            return 1024*1024*100;
```

Comment:
This seems extremely low?

Reply:
`totalMemory` isn't system memory, but - in my understanding - memory currently reserved for FP specifically. This grows over time as FP allocates more.
For example in FP's root DO constructor, I'm getting:
```
privateMemory:
10 825 728
totalMemory:
5 398 528
totalMemoryNumber:
5 439 488
freeMemory:
2 744 320
```
So I'd say it's actually more than you'll see in an average game.

Reply:
Oh, that is on me for not reading the documentation. I do think privateMemory should probably be a bit higher than totalMemory, just based on this quote for totalMemory:
> This property does not return all memory used by an Adobe AIR application or by the application (such as a browser) containing Flash Player content. The browser or operating system may consume other memory. The System.privateMemory property reflects all memory used by an application.